src_python/habitat_sim/simulator.py

```diff
@@ -431,18 +437,23 @@ def get_sensor_observations(
         else:
             return_single = False
 
-        for agent_id in agent_ids:
-            agent_sensorsuite = self.__sensors[agent_id]
-            for _sensor_uuid, sensor in agent_sensorsuite.items():
-                sensor.draw_observation()
-
-        # As backport. All Dicts are ordered in Python >= 3.7
+        # As backport. All Dicts are ordered in Python >= 3.7.
         observations: Dict[int, ObservationDict] = OrderedDict()
+
+        # Draw observations for classic non-batched renderer.
+        if not self.config.enable_batch_renderer:
+            for agent_id in agent_ids:
+                agent_sensorsuite = self.__sensors[agent_id]
+                for _sensor_uuid, sensor in agent_sensorsuite.items():
+                    sensor.draw_observation()
+
+        # Get observations.
         for agent_id in agent_ids:
```

Comment:
Can you explain what's going on here? How can we safely call `get_observation` if we haven't rendered here? Is this relying on some external code to have already triggered rendering?

Reply:
When batch rendering, `get_observation` returns `None`, creating an empty "placeholder" for this sensor uuid. The batch renderer replaces this value by the rendered data in `post_step()`.

This is a bit of a hack as Habitat-Lab has checks to make sure that the sensor uuid is present between `reset()`/`step()` and `post_step()`.